### ISSUE TITLE
LSP: disable wrapping in hover window

### ISSUE THREAD

**Issue by ndavd** [neovim/neovim#30831]
### Problem

Context: https://github.com/juanfranblanco/vscode-solidity/issues/467.

Some LSPs display the hover text wrapped:
![Image](https://github.com/user-attachments/assets/d8460692-f24f-4cc2-bbee-358ca28c3ded)

The solution would be to do `set nowrap` automatically, as mentioned here https://github.com/juanfranblanco/vscode-solidity/issues/467#issuecomment-2416302302.

### Expected behavior

The text should not wrap, the window should instead have dynamic calculated dimensions according to the text.

**Comment by MariaSolOs** [comment]
You can always override the hover handler and disable wrapping with its configuration argument. [The docs show how to do this](https://neovim.io/doc/user/lsp.html#vim.lsp.handlers.hover()).

Moreover are you sure that the line breaks are added by NeoVim and not sent by the server? [Here's](https://github.com/neovim/neovim/blob/395f420fc65cb99f1c5cf17e230b962cb1e74bcd/runtime/lua/vim/lsp/util.lua#L1458) where the dimensions of the floating window are computed, and although I haven't debugged/analyzed it carefully I think it shouldn't create the kind of wrapped text shown above.

**Comment by ndavd** [comment]
> You can always override the hover handler and disable wrapping with its configuration argument. [The docs show how to do this](https://neovim.io/doc/user/lsp.html#vim.lsp.handlers.hover()).

If I add `wrap = false`, then the window is too small for the text and it gets cut by it, i.e. in order to see the end of the sentence I must enter the window and scroll horizontally. Does this confirm it's an LSP issue and not a neovim one?

**Comment by justinmk** [comment]
> I think it shouldn't create the kind of wrapped text shown above.

The "weird" wrapping is because of 'breakindent' (which I added in https://github.com/neovim/neovim/pull/30695 ). That's tradeoff but disabling it would only change where the text is wrapped to...

**Comment by ndavd** [comment]
> Your screenshot doesn't show the surrounding layout, so we can only guess whether the dimensions were decided correctly in this case.

I should've pasted a bigger screenshot indeed. Here's from left to right of the window.![Image](https://github.com/user-attachments/assets/d32a7318-c05a-4cbf-85f8-97133ea7cd9b)

**Comment by justinmk** [comment]
Yeah, I would expect the window to be wider. Possibly a bug in the logic that calculates that.

**Comment by glepnir** [comment]
can't reproduce now

<img width="1996" height="628" alt="Image" src="https://github.com/user-attachments/assets/810dc80b-f38c-491a-ad56-0d322a1bd100" />

**Comment by fnky** [comment]
Stumped into this problem as well since a long time ago. Here's just a simple example where the hover cuts off 2 characters at the width:

<img width="404" height="153" alt="Image" src="https://github.com/user-attachments/assets/b0e52873-5eed-4e07-a0c5-6f89f54015f3" />

Jumping into the hover to scroll to the end of the line:

<img width="390" height="137" alt="Image" src="https://github.com/user-attachments/assets/49d1853b-fd29-4205-8956-3574d76cd83a" />

I have no clue how I would be able to fix this in user-land, since setting `width`/`height` doesn't account for text length/paddings of the content. I'd rather not copy the whole built-in method to fix this tiny issue but is currently the only solution I can think of.

**Comment by glepnir** [comment]
can't reproduce on master

<img width="1134" height="470" alt="Image" src="https://github.com/user-attachments/assets/c51a8e98-3e38-4358-b901-b2d686e57fdb" />

**Comment by justinmk** [comment]
@fnky always mention the exact version of Nvim you are using and repro steps with `nvim --clean`.

**Comment by lisiur** [comment]
I ran into the same issue. I found out it was caused by the render-markdown.nvim plugin because I had code.left_pad = 2 enabled.



> Stumped into this problem as well since a long time ago. Here's just a simple example where the hover cuts off 2 characters at the width:
> 
> <img alt="Image" width="404" height="153" src="https://private-user-images.githubusercontent.com/995050/478395707-b0e52873-5eed-4e07-a0c5-6f89f54015f3.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzcyNTk1MjEsIm5iZiI6MTc3NzI1OTIyMSwicGF0aCI6Ii85OTUwNTAvNDc4Mzk1NzA3LWIwZTUyODczLTVlZWQtNGUwNy1hMGM1LTZmODlmNTQwMTVmMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNDI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDQyN1QwMzA3MDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hYTEyNmQzNDAwYTBhNjM2M2FiMzg3M2QyYjJmMDJkODUwOWFkNGVkNTUxYjZjNmEzMGQyZjllNGMwYzE1OTFiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.nX-WQn4Tiommtj3fT74tqYHgaai5Mnw0BNIM_M1aXoU">
> Jumping into the hover to scroll to the end of the line:
> 
> <img alt="Image" width="390" height="137" src="https://private-user-images.githubusercontent.com/995050/478400063-49d1853b-fd29-4205-8956-3574d76cd83a.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzcyNTk1MjEsIm5iZiI6MTc3NzI1OTIyMSwicGF0aCI6Ii85OTUwNTAvNDc4NDAwMDYzLTQ5ZDE4NTNiLWZkMjktNDIwNS04OTU2LTM1NzRkNzZjZDgzYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNDI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDQyN1QwMzA3MDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04NmIxOTMwMzlmYTNjYTI3ZjNmYzg4Y2UzMWMxYTNlNTM1NTY3MjIyNThmOWI2ZTZhNDUwODhjMTQ0MjYxYzhmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.CJZ5jn5IFXZnSetRy2ZVb0peL-qM4Ry3RHAHM6I9eQ8">
> I have no clue how I would be able to fix this in user-land, since setting `width`/`height` doesn't account for text length/paddings of the content. I'd rather not copy the whole built-in method to fix this tiny issue but is currently the only solution I can think of.

Based on your screenshot, it seems you’re also using a similar plugin for line numbers. It takes up a width of exactly 2 characters.

PS:
```
NVIM v0.12.2
Build type: Release
LuaJIT 2.1.1774896198
```
